rules/coding-style/src/Rector/ClassMethod/RemoveParamReturnDocblockRector.php

```diff
@@ -95,101 +91,39 @@ public function getNodeTypes(): array
-        $phpDocInfo = $classMethod->getAttribute(AttributeKey::PHP_DOC_INFO);
-        if ($phpDocInfo === null) {
-            throw new ShouldNotHappenException();
-        }
-
-        /** @var Type[] $paramTypes */
-        $paramTypes = $phpDocInfo->getParamTypesByName();
+        foreach ($phpDocInfo->getParamTagValueNodes() as $paramTagValueNode) {
+            if (! $this->deadParamTagValueNodeAnalyzer->isDead($paramTagValueNode, $classMethod)) {
+                continue;
+            }
 
-        /** @var Param[] $params */
-        $params = $classMethod->getParams();
-        foreach ($paramTypes as $key => $type) {
-            $docCommentText = $this->processCleanDocCommentParam($params, $key, $type, $docCommentText);
+            $phpDocInfo->removeTagValueNodeFromNode($paramTagValueNode);
         }
-
-        return $docCommentText;
     }
 
-    private function processUpdateDocblock(ClassMethod $classMethod, string $text, string $docCommentText): ?ClassMethod
+    private function refactorReturnTag(ClassMethod $classMethod, PhpDocInfo $phpDocInfo): void
     {
-        if ($docCommentText === $text) {
-            return null;
+        $attributeAwareReturnTagValueNode = $phpDocInfo->getReturnTagValue();
+        if ($attributeAwareReturnTagValueNode === null) {
+            return;
         }
 
-        $classMethod->setDocComment(new Doc($docCommentText));
-        $expressionPhpDocInfo = $this->phpDocInfoFactory->createFromNode($classMethod);
-        $classMethod->setAttribute(AttributeKey::PHP_DOC_INFO, $expressionPhpDocInfo);
-
-        return $classMethod;
-    }
-
-    /**
-     * @param array<int, Param> $params
-     */
-    private function processCleanDocCommentParam(array $params, string $key, Type $type, string $docCommentText): string
-    {
-        foreach ($params as $param) {
-            $paramName = '';
-            $paramVarName = $this->getName($param->var);
-
-            if ($key !== '$' . $paramVarName) {
-                continue;
-            }
-
-            if ($param->type instanceof FullyQualified && ($type instanceof ShortenedObjectType || $type instanceof FullyQualifiedObjectType)) {
-                $paramName = addslashes($type->getClassName());
-            }
-
-            if ($param->type instanceof Identifier) {
-                $paramName = $param->type->toString();
-            }
-
-            if ($paramName === '') {
-                continue;
-            }
-
-            $paramRegex = sprintf(self::PARAM_REGEX, $paramName, $paramVarName);
-            if (Strings::match($docCommentText, $paramRegex)) {
-                $docCommentText = Strings::replace($docCommentText, $paramRegex, '');
-            }
+        if (! $this->deadReturnTagValueNodeAnalyzer->isDead($attributeAwareReturnTagValueNode, $classMethod)) {
+            return;
         }
 
-        return $docCommentText;
+        $phpDocInfo->removeTagValueNodeFromNode($attributeAwareReturnTagValueNode);
     }
```

Comment:
@samsonasik I'm working on removing of direct use of `getDocComment()` and `getComments()`, and as one step I've refactorted this rule to PHPStan types. It might be good guide on how to work with do comments in the future.

Reply:
Thank you